docs/dev.md

```diff
@@ -0,0 +1,420 @@
+# Development
+
+You can build, check, and test dcos-net in a development image using
+`make docker-compile`, `make docker-check`, and `make docker-test` respectively.
+All makefile targets with `docker-` prefix build development image with all
+dependencies and run `rebar3` in that image on the host directory.
+
+To check your dcos-net build on DC/OS you can use DC/OS E2E a.k.a.
+[dcos-docker CLI](http://dcos-e2e.readthedocs.io/en/latest/cli.html):
+
+```sh
+curl -O https://downloads.dcos.io/dcos/testing/master/dcos_generate_config.sh
+make dcos-docker-create DCOS_DOCKER_AGENTS=3
+make dcos-docker-dev DCOS_DOCKER_NODE=master_0
+make dcos-docker-shell DCOS_DOCKER_NODE=agent_1
+make dcos-docker-destroy
+```
+
+Alternatively, here are instructions on how to run everything manually. Going
+forward, it will be assumed that there are two machines available, `node-1` and
+`node-2` respectively. They can be physical machines, virtual machines, or even
+Docker containers. Let's assume that their IPv4 addressed are:
+
+```sh
+NODE_1_IP=192.168.0.1
+NODE_2_IP=192.168.0.2
+```
+
+Firstly, the dependencies should be installed. Most of them can be installed
+using the package manager of your Linux distribution. In case Erlang 21.x is not
+available through it, please build one using [kerl](https://github.com/kerl/kerl):
+
+```sh
+KERL_CONFIGURE_OPTIONS=
+KERL_CONFIGURE_OPTIONS+="--enable-dirty-schedulers "
+KERL_CONFIGURE_OPTIONS+="--enable-kernel-poll "
+KERL_CONFIGURE_OPTIONS+="--with-hipe"
+KERL_CONFIGURE_OPTIONS+="--with-ssl"
+KERL_CONFIGURE_OPTIONS+="--without-javac "
+export KERL_CONFIGURE_OPTIONS
+
+kerl build 21.1 21.1
+kerl install 21.1 $HOME/erl
+```
+
+Once it is done, you may execute the following command to activate it:
+
+```sh
+source $HOME/erl/activate
+```
+
+Next step is to build Mesos itself. It is really easy to do so using
+[Mesos Version Manager](https://github.com/mesosphere/marathon/blob/master/tools/mvm.sh).
+In order to install it, run:
+
+```sh
+curl -LO https://raw.githubusercontent.com/mesosphere/marathon/master/tools/mvm.sh
+chmod +x mvm.sh
+```
+
+And build Mesos using it:
+
+```sh
+mvm.sh 1.7.0
```

Comment:
Maybe it pays of to comment that compiling mesos have a lot of dependencies, and point to this docs, because it helps you quickly find the dependencies for your platform (http://mesos.apache.org/documentation/latest/building/).